### DISCUSSION TITLE
[Question] Please Help me!!  How can I check end time when it ended playing ft.Video?

### DISCUSSION THREAD

**Discussion by jjjunehkim** [flet-dev/flet#3042] [Q&A]
### Question

Please Describe The Problem To Be Solved

**I want to making viewer and player with both image files and mp4 files.**

1. first, viewing image files every 1.3 seconds   
2. after end of showing image files, start playing mp4 files with ft.Video
3. after end of playing mp4 files, start playing viewing image file again
**I want these tasks(1, 2, 3) to be infinitely repeated**

so I need to check the time after mp4 play ends for playing viewer with image files after mp4 play ends, 
I tried to find all of "Properties" and "Methods" in  flet.dev, but I cannot find it.

Operating system: Windows 11 

### ------------------------------------------------------

- [x] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by wschume** [comment]
How about using `async` with some tasks (see [here](https://flet.dev/docs/getting-started/async-apps/)). Then a combination of [`is_completed`](https://flet.dev/docs/controls/video#is_completed) and [`is_playing`](https://flet.dev/docs/controls/video#is_completed) should do the trick.

**Reply by jjjunehkim** [reply]
Thank you so much for your answering!
and Sorry for my misspell(it is Not mp3 file, it is mp4 file). I corrected it

I checked this method(->[is_completed]) how to work.
I know [is_completed] Method returns true or false just about one file. 
So, Even though remaining another mp4 file for playing in playlist, 
when it is the end of the play the first mp4 file,  [is_completed] Method returns true regardless of using async,

**Reply by wschume** [reply]
Maybe you should post a [minimum working example](https://stackoverflow.com/help/minimal-reproducible-example). This way it easier to see what you are doing, and also to maybe tweak your code into the desired behavior. Make it as minimal as possible ^^.

**Reply by jjjunehkim** [reply]
Sorry about that I didn't post example code. 
Please understand my situation. 
Now I am just readying before starting do code. So I don't have my own code for posting here. 

I need to find the solution before starting. 
After checking the end time of the play through partial changing and modifying code with sample in flet.dev,
I am going to start.
